File: testcpp.ts
/**
 * sample cpp package
 */
//% color=156 weight=80
declare namespace testcpp {

  /**
   * Says hello world, as defined in the ts file
   */
  //% weight=30 help=testcpp/helloWorld
  //% blockId=testcpp_block icon="\uf030" block="say hello world|%property" blockGap=8 shim=testcpp::helloWorld
  function helloWorld(duration: TestcppParams): void;
}

File: testcpp.cpp
#include "pxt.h"

using namespace pxt;

enum class TestcppParams {
  //% block = "forever"
  Forever = 0,
  //% block = "for now"
  ForNow = 1,
};

/**
 * Sample comment block
 */
//% color=156 weight=80
namespace testcpp {

  /**
   * Sample hello world function
   */
  //% weight=30 help=testcpp/helloWorld
  //% blockId=devices_camera icon="\uf030" block="helloWorld|%property" blockGap=8
  void helloWorld(TestcppParams duration) {
  }
}
